web/views.py

```
@@ -2736,7 +2754,133 @@ def challenge_submit(request, week_number):
+    if not quiz:
+        messages.error(request, "The requested quiz was not found or is no longer active.")
+        return redirect("index")
+
+    user_submission = None
+    has_submitted = False
+
+    # Fetch questions only if needed
+    questions = quiz.questions.all().prefetch_related("options")
+    paginator = Paginator(questions, 10)  # 10 questions per page
+    page_number = request.GET.get("page", 1)
+    questions_page = paginator.get_page(page_number)
```

Ruff (0.8.2)

Local variable `questions_page` is assigned to but never used

Remove assignment to unused variable `questions_page`

(F841)